### PR TITLE
Devdocs 1274 paymentsapi vaulted paypal

### PR DESCRIPTION
**Pull request by mikrdgz**
# [DEVDOCS-1274](https://jira.bigcommerce.com/browse/DEVDOCS-1274)

## What changed?
Adds paypal instrument and supported instrument to Payments API GET, and POST to /payments accepts stored paypal account

### CODE REVIEW

**Comment by aglensmith** [comment]
Need to compare this against engineerings spec file -- did jack give you an updated one?

**Comment by mikrdgz** [comment]
> Need to compare this against engineerings spec file -- did jack give you an updated one?

https://github.com/bigcommerce/api-schema/blob/master/bigpay/v1/docs/payment-api.yaml

https://github.com/bigcommerce/bigcommerce/blob/master/docs/InternalApi/V1/Payments/payments.yaml

File: reference/json/BigCommerce_Payments_API.oas2.json [hunk 1]
```diff
@@ -1375,7 +1375,50 @@
           }
         }
       }
+    },
+    "StoredPaypalInstrument": {
```
Comment (aglensmith) [inline reference/json/BigCommerce_Payments_API.oas2.json]: I don't see `StoredPaypalInstrument` in the schema that you linked. I think this should be changed to `StoredPayPalAccountInstrument`

File: reference/json/BigCommerce_Process_Payment_API.oas2.json [hunk 2]
```diff
@@ -652,8 +652,27 @@
...
         "type"
       ]
-    }
-  },
+    },
+  "StoredPayPalAccount": {
+    "title": "Stored Paypal Account",
+    "type": "object",
+    "properties": {
+      "type": {
```
Comment (aglensmith) [inline reference/json/BigCommerce_Process_Payment_API.oas2.json]: The `type` property here needs to be set as an enum with `stored_paypal_account` as the only option: 

```yaml
    StoredPayPalAccount:
      type: object
      properties:
        type:
          type: string
          description: "Type to classify this payment instrument"
          enum:
            - stored_paypal_account
```

File: reference/json/BigCommerce_Process_Payment_API.oas2.json [hunk 3]
```diff
@@ -652,8 +652,27 @@
...
+      "type": "string",
+      "minLength": 64,
+      "maxLength": 64
+        }
+      },
+    "required": [
+    "type",
+    "token"
+    ]
+  }
```
Comment (aglensmith) [inline reference/json/BigCommerce_Process_Payment_API.oas2.json]: Probably also need to add `PaymentInstrument` object to definitions and include `StorePayPalAccount` reference: 

```yaml
    PaymentInstrument:
      oneOf:
        - $ref: "#/components/schemas/Card"
        - $ref: "#/components/schemas/StoredCard"
        - $ref: "#/components/schemas/StoredPayPalAccount"
      discriminator:
          propertyName: type
          mapping:
            card: "#/components/schemas/Card"
            stored_card: "#/components/schemas/StoredCard"
            stored_paypal_account: "#/components/schemas/StoredPayPalAccount"
```

**Comment by jackosaurus** [comment]
Hey guys, should I still review this one? Let me know, thanks!

**Pull request closed without merging**